apis/common/v1/condition.go

```diff
@@ -141,6 +157,9 @@ func (s *ConditionedStatus) SetConditions(c ...Condition) {
 
 			if existing.Equal(new) {
```

Comment:
GitHub won't let me add a comment to the `existing.Equal` docstring, but it should explicitly call out that it doesn't consider `ObservedGeneration` when assessing equality. It already calls out that it ignores LastTransitionTime.